src/lib_json/json_writer.cpp

```diff
@@ -141,14 +142,15 @@ std::string valueToString(double value, bool useSpecialFloats, unsigned int prec
   char buffer[32];
   int len = -1;
 
-  char formatString[6];
-  sprintf(formatString, "%%.%dg", precision);
+  std::stringstream value_ss;
+  value_ss << std::setprecision( (int)precision ) << value;
```

Comment:
clang complains about this cast, not me.
I'll change to static_cast asap.

2016-02-12 16:55 GMT+01:00 Billy Donahue <notifications@github.com>:

> In src/lib_json/json_writer.cpp
> <https://github.com/open-source-parsers/jsoncpp/pull/421#discussion_r52758164>
> :
>
> > @@ -141,14 +142,15 @@ std::string valueToString(double value, bool useSpecialFloats, unsigned int prec
> >    char buffer[32];
> >    int len = -1;
> >
> > -  char formatString[6];
> > -  sprintf(formatString, "%%.%dg", precision);
> > +  std::stringstream value_ss;
> > +  value_ss << std::setprecision( (int)precision ) << value;
>
> This cast isn't necessary. The precision must have a value small enough to
> be representable as an int. If that condition holds, this is a well-defined
> conversion without the cast. If this condition does not hold, then then
> this cast is invalid. The cast doesn't provide any value as far as I can
> see. If you need it for some reason (compiler warnings?), it's better as
> static_cast(precision), which is a weaker cast than C-style casting.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/open-source-parsers/jsoncpp/pull/421/files#r52758164>.
>



-- 
Fiorentino Ing. Stefano
Cell. +39 340 2216385